src/app/clusters/scenes-server/scenes-server.cpp

```diff
@@ -0,0 +1,993 @@
+        (0 != req.groupIdentifierTo &&
+         !mGroupProvider->HasEndpoint(ctx.mCommandHandler.GetAccessingFabricIndex(), req.groupIdentifierTo, mPath.mEndpointId)))
+    {
+        response.status = to_underlying(Protocols::InteractionModel::Status::InvalidCommand);
+        ctx.mCommandHandler.AddResponse(mPath, response);
+        return;
+    }
+
+    uint8_t capacity = 0;
+    // Get Capacity
+    ReturnOnFailure(AddResponseOnError<Commands::CopySceneResponse::Type>(
+        ctx, response, sceneTable->GetRemainingCapacity(ctx.mCommandHandler.GetAccessingFabricIndex(), capacity)));
+
+    // Checks if we copy a single scene or all of them
+    if (req.mode.GetField(app::Clusters::Scenes::ScenesCopyMode::kCopyAllScenes))
+    {
+        // Scene Table interface data
+        SceneId scenesInGroup[scenes::kMaxScenesPerFabric];
+        Span<SceneId> sceneList = Span<SceneId>(scenesInGroup);
+
+        // populate scene list
+        ReturnOnFailure(AddResponseOnError<Commands::CopySceneResponse::Type>(
+            ctx, response,
+            sceneTable->GetAllSceneIdsInGroup(ctx.mCommandHandler.GetAccessingFabricIndex(), req.groupIdentifierFrom, sceneList)));
+
+        if (0 == capacity)
+        {
+            response.status = to_underlying(Protocols::InteractionModel::Status::ResourceExhausted);
+            ctx.mCommandHandler.AddResponse(mPath, response);
+            return;
+        }
+
+        for (auto & sceneId : sceneList)
+        {
+            SceneTableEntry scene(SceneStorageId(sceneId, req.groupIdentifierFrom));
+            //  Insert in table
+            ReturnOnFailure(AddResponseOnError<Commands::CopySceneResponse::Type>(
+                ctx, response,
+                sceneTable->GetSceneTableEntry(ctx.mCommandHandler.GetAccessingFabricIndex(), scene.mStorageId, scene)));
+
+            scene.mStorageId = SceneStorageId(sceneId, req.groupIdentifierTo);
+
+            ReturnOnFailure(AddResponseOnError<Commands::CopySceneResponse::Type>(
+                ctx, response, sceneTable->SetSceneTableEntry(ctx.mCommandHandler.GetAccessingFabricIndex(), scene)));
+        }
+
+        response.status = to_underlying(Protocols::InteractionModel::Status::Success);
+        ctx.mCommandHandler.AddResponse(mPath, response);
+        return;
+    }
+
+    SceneTableEntry scene(SceneStorageId(req.sceneIdentifierFrom, req.groupIdentifierFrom));
+    ReturnOnFailure(AddResponseOnError<Commands::CopySceneResponse::Type>(
+        ctx, response, sceneTable->GetSceneTableEntry(ctx.mCommandHandler.GetAccessingFabricIndex(), scene.mStorageId, scene)));
+
+    scene.mStorageId = SceneStorageId(req.sceneIdentifierTo, req.groupIdentifierTo);
+
+    ReturnOnFailure(AddResponseOnError<Commands::CopySceneResponse::Type>(
+        ctx, response, sceneTable->SetSceneTableEntry(ctx.mCommandHandler.GetAccessingFabricIndex(), scene)));
+
+    // Update Attributes
```

Comment:
Again, we shouldn't be doing that.  We should just be marking them dirty.

Importantly: we should be marking them dirty in the "copy all scenes" case above too!